### PR TITLE
Initial cleanup of web_character.py

### PR DESCRIPTION
**Pull request by BarinderBanwait**
First step towards LMFDB #4172

### CODE REVIEW

File: lmfdb/characters/test_characters.py [hunk 1]
```diff
@@ -116,11 +106,11 @@ def test_dirichletcharbig(self):
             This test also makes sure the code scales a little bit.
         """
         W = self.tc.get('/Character/Dirichlet/40487/5')
-        assert '40486' in W.get_data(as_text=True)
-        assert '12409' in W.get_data(as_text=True), "log on generator"
```
Comment (BarinderBanwait) [inline lmfdb/characters/test_characters.py:120]: @AndrewVSutherland Since we're no longer using `WebDirichletCharacter` for modulus between 10,000 and 100,000, the `Values on generators` doesn't appear on this page, so this test failed. I therefore changed it to check the field of values; but do let me know if I misunderstood something, or if the behaviour should be otherwise.

File: lmfdb/characters/test_characters.py [hunk 2]
```diff
@@ -169,5 +159,5 @@ def test_dirichletchar9999lfunc(self):
     def test_dirichletchar99999999999999999lfunc(self):
         """ Check Dirichlet character with very large modulus"""
         W = self.tc.get('/Character/Dirichlet/99999999999999999999/2')
-        assert parity_string(-1) in W.get_data(as_text=True) and '536870912' in W.get_data(as_text=True)
```
Comment (BarinderBanwait) [inline lmfdb/characters/test_characters.py:172]: Since we're removing the `Galois orbit` section in the large modulus case, this test failed, so as above I changed it to something that *is* there. Again do let me know if it should be otherwise.

File: lmfdb/characters/test_characters.py (hunk 1 above)
Comment (AndrewVSutherland) [reply lmfdb/characters/test_characters.py:120]: Actually, I realized that we can keep the values on generators and the values, because Pari now supports Conrey characters and can compute these very efficiently.  To compute the value of chi_q(m,n) you can use:

```
 def conreyangle(q,m,n):
     return Pari("G=znstar(%d,1); chareval(G,znconreylog(G,%d),%d)"%(q,m,n))
```

which will return a rational number r such that chi_q(m,n) = exp(2*pi*r).  This is very fast even for 10 or 20 digit values of q.

To get generators you can use `Integers(q).unit_gens()`.

With this we can easily keep the values on generators and values sections for all Dirichlet characters, as well as the ability to compute values, just as we do for characters in the database.

File: lmfdb/characters/test_characters.py (hunk 2 above)
Comment (AndrewVSutherland) [reply lmfdb/characters/test_characters.py:172]: I agree with removing the Galois orbit section for characters of modulus > 10000 which are not in the database, since we don't actually know the orbit label.

**Comment by AndrewVSutherland** [comment]
One other comment, I think we should just use Pari to compute everything about Conrey characters (we only need to know how to compute the conductor, order, parity, isreal, and the inducting primitive character), all of which Pari can do quite efficiently).  This would let us remove the dependence on the dirichlet_conrey package, which has always been a headache.  See https://pari.math.u-bordeaux.fr/dochtml/html/Arithmetic_functions.html for documentation.

**Comment by JohnCremona** [comment]
I agree that using pari for d.c. chars is a good idea. (I discovered recently that a trivial intervention is needed to pip install Bober's package in sage 9.2 anyway.)

To avoid multiple rather arcane calls as in Drew's earlier comment it would nice if someone wrote a few little wrapper functions for what we'll need.

File: lmfdb/characters/test_characters.py (hunk 1 above)
Comment (BarinderBanwait) [reply lmfdb/characters/test_characters.py:120]: OK thanks for the feedback @AndrewVSutherland , I'll aim to get `Values on generators` as well as `First values` back onto the pages for large modulus via Pari.

**Comment by BarinderBanwait** [comment]
OK sounds good - I'll aim to replace all the dirichlet_conrey stuff with direct calls to the `znconrey` methods, complete
with all the pari wrapper functions we'd need. :toolbox:

File: lmfdb/verify/mf_newspaces.py [hunk 3]
```diff
@@ -330,7 +341,7 @@ def check_dims(self, rec, verbose=False):
         if rec['level'] < 3:
             dirchar = rec['level']
         else:
-            dirchar = DirichletGroup_conrey(rec['level'])[rec['conrey_indexes'][0]].sage_character()
+            dirchar = get_dirchar(rec['level'], rec['conrey_indexes'][0], rec['char_order'])
```
Comment (BarinderBanwait) [inline lmfdb/verify/mf_newspaces.py:344]: test

File: lmfdb/verify/mf_newspaces.py [hunk 4]
```diff
@@ -318,10 +329,10 @@ def check_Skchi_dim_formula(self, rec, verbose=False):
...
```
Comment (BarinderBanwait) [inline lmfdb/verify/mf_newspaces.py:332]: The changes in this file - here and below  - are untested, since I didn't know how to test them. I'd be grateful if someone could show me how to run these verification scripts.

**Comment by BarinderBanwait** [comment]
OK, so I've removed the dirichlet_conrey package as indicated above; most of the Pari calls went into the existing `ConreyCharacter` class in the `TinyConrey` module (a few are in a new `PariConreyGroup` class in the same module; this is a bare-bones analogue of the previous `DirichletGroup_conrey`).

This PR has snowballed somewhat, so I feel an overview of what it's now doing is in order.

The main user-facing changes are as follows:

1. Galois orbits have been removed for modulus >=10,000.
2.  Values on Generators, and the Values/First Values table have been added for modulus >=100,000.
3. The label of the induced character for modulus >= 100,000 is now being computed and is showing (previously it said 'None'; e.g. [here](https://www.lmfdb.org/Character/Dirichlet/100001/23)).
4. Compute buttons for Values/Gauss/Jacobi/Kloosterman sums now work (previously they did nothing).
5. The SageMath code stubs to compute the character now explicitly specifies the base ring; not doing this can sometimes yield a ValueError when running the previous code in sage (e.g. for 91.3; the database considers the multiplicative order as 6, but just calling DirichletGroup(91) yields base_ring as CyclotomicField(12), so the call to DirichletCharacter was failing; this was noticed when fixing unit test failures).

The other non-user-facing changes are that web_character has been cleaned up a bit (which was the original goal), and some tests have been added.

The `dirichlet_conrey` package was being used in the verify directory, which I don't know anything about. I've flagged these points as comments in this PR, since I haven't been able to test these changes.

**Review by AndrewVSutherland — APPROVED** [review]

**Comment by AndrewVSutherland** [comment]
It looks like the 10000 cutoff is off by 1, we have characters of modulus <= 10,000 in the database, but code seems to think we only have them up to < 10,000.

**Comment by AndrewVSutherland** [comment]
Also the values table is a bit wide for the characters of modulus > 10,000, e.g. 

http://127.0.0.1:37777/Character/Dirichlet/10001/7

Now the we have the ability to compute arbitrary values on these pages, I think just showing, say, the first 10 values in addition to the value at -1 would be enough.

**Comment by AndrewVSutherland** [comment]
Looks good, happy to merge once tests pass.

**Comment by BarinderBanwait** [comment]
Latest commit addresses the two issues raised in the last two comments.

@AndrewVSutherland David Roe showed me how to run the verification scripts, but that test I changed is not passing, and it looks like it might go back to an issue with constructing the sage character; so until I've got to the bottom of that it's probably best to hold-off merging this PR.

File: lmfdb/verify/mf_newspaces.py (hunk 4 above)
Comment (BarinderBanwait) [reply lmfdb/verify/mf_newspaces.py:332]: Thanks @roed314 for showing me how to run these (and fixing them along the way)! The verification is not passing right now, which I'm getting to the bottom of

Comment (BarinderBanwait) [reply lmfdb/verify/mf_newspaces.py:332]: Latest commit c321e21 resolves the verification test failures. For details of the issue see my comment below that commit.

**Comment by BarinderBanwait** [comment]
TL;DR: Verification tests now pass. @AndrewVSutherland  I believe this is good to be merged now.

More details:

There was an issue with the way I was constructing the sage character. I based it off the sage generating code which shows when you click "Show commands for: SageMath". The list of genvalues corresponds to the numerators of the fractions that appear in the values on generators. But these values can't be used verbatim when constructing the character in sage; they need to be adjusted to account for the denominator; in the db this is the displayed order, but in sage this is `H.zeta_order()`, and they can well be different. This meant that the gauss/jacobi/kloosterman sums might be wrong.

This adjustment was being done in Bober's package, which is how the verification tests were passing, so I took a look at his code and copied his logic into a new `get_sage_genvalues` method.

The two verification tests then passed; in the interest of time I set ratio as `float(0.0001)`, which corresponded to 127 test cases; `check_Skchi_dim_formula` passed after 43 seconds, and `check_dims` passed in about 900 seconds.

The various examples that were failing while debugging the verification test failures have now been encoded into a new unit test `test_sage_code_gens`, to ensure correct sage character in future.
